### PR TITLE
Misc cleanup of VM threading code

### PR DESCRIPTION
**Pull request by jkotas**
- Delete special-casing of thread finalizer
- Factor out Get/SetApartment of unstarted threads
- Delete some dead code
- Fix comments

### CODE REVIEW

File: src/coreclr/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs [hunk 1]
```diff
@@ -240,10 +240,7 @@ public ThreadPriority Priority
             {
                 Thread _this = this;
                 SetPriority(ObjectHandleOnStack.Create(ref _this), (int)value);
-                if (value != ThreadPriority.Normal)
-                {
-                    _mayNeedResetForThreadPool = true;
-                }
+                _mayNeedResetForThreadPool = true;
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs:243]: This was not handling the case of finalizer threads.

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Thread.NativeAot.Windows.cs [hunk 2]
```diff
@@ -14,9 +14,6 @@ namespace System.Threading
 {
     public sealed partial class Thread
     {
-        [ThreadStatic]
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Thread.NativeAot.Windows.cs:17]: Dead code

File: src/coreclr/vm/finalizerthread.cpp [hunk 3]
```diff
@@ -426,19 +376,8 @@ DWORD WINAPI FinalizerThread::FinalizerThreadStart(void *args)
...
     ASSERT(hEventFinalizer->IsValid());
 
-    // TODO: The following line should be removed after contract violation is fixed.
-    // See bug 27409
-    SCAN_IGNORE_THROW;
-    SCAN_IGNORE_TRIGGER;
-
     LOG((LF_GC, LL_INFO10, "Finalizer thread starting...\n"));
 
-#if defined(FEATURE_COMINTEROP_APARTMENT_SUPPORT) && !defined(FEATURE_COMINTEROP)
```
Comment (jkotas) [inline src/coreclr/vm/finalizerthread.cpp:436]: The condition was always false.

File: src/coreclr/vm/threads.cpp [hunk 4]
```diff
@@ -4799,6 +4770,9 @@ Thread::ApartmentState Thread::SetApartment(ApartmentState state)
     }
     CONTRACTL_END;
 
+    // This method can be called on current thread only
```
Comment (jkotas) [inline src/coreclr/vm/threads.cpp:4773]: Fixed the non-current thread callers to use SetApartmentOfUnstartedThread

**Comment by jkotas** [comment]
Motivated by HMF->QCall conversion codereviews

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Comment by JulieLeeMSFT** [comment]
@jkotas, all test legs failed in [runtime-coreclr libraries-jitstress](https://dev.azure.com/dnceng-public/public/_build?definitionId=138&_a=summary)/[20240924.1](https://dev.azure.com/dnceng-public/public/_build/results?buildId=816649). PTAL.

**Comment by jkotas** [comment]
I have looked at some of the failures. The failures are either asserts in the JIT

` Assertion failed '!"Inconsistent profile data"' in 'Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer:LexXmlDocCommentLeadingTrivia(byref):this' during 'Recognize Switch' (IL size 730; hash 0xe3243bf9; Tier1)`

https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-heads-main-a4b3f25a20cf46ce8d/System.Text.RegularExpressions.Tests/1/console.2ae64c75.log?skoid=8eda00af-b5ec-4be9-b69b-0919a2338892&sktid=72f988bf-86f1-41af-91ab-2d7cd011db47&skt=2024-09-24T18%3A02%3A37Z&ske=2024-09-24T19%3A02%3A37Z&sks=b&skv=2024-05-04&sv=2024-05-04&st=2024-09-24T18%3A02%3A37Z&se=2024-09-24T19%3A02%3A37Z&sr=c&sp=r&sig=e35ir71XB%2BNVh0DzOhQSt7pwGDj9BrlFxUC2QGob%2BAk%3D

Or asserts/crashes in libraries that suggest bad codegen:

https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-heads-main-5f622e8df388454594/System.Text.Json.Tests/1/console.dbf8d227.log?skoid=8eda00af-b5ec-4be9-b69b-0919a2338892&sktid=72f988bf-86f1-41af-91ab-2d7cd011db47&skt=2024-09-24T18%3A04%3A02Z&ske=2024-09-24T19%3A04%3A02Z&sks=b&skv=2024-05-04&sv=2024-05-04&st=2024-09-24T18%3A04%3A02Z&se=2024-09-24T19%3A04%3A02Z&sr=c&sp=r&sig=bL%2BAUD%2Bxfmt5N6ov%2F8w6uwftzRUy0AkdDHRY%2FSGXwbI%3D

**Comment by jkotas** [comment]
Similar set of failures was hit in https://dev.azure.com/dnceng-public/public/_build/results?buildId=815402&view=ms.vss-test-web.build-test-results-tab that did not include this change.

**Comment by JulieLeeMSFT** [comment]
> I have looked at some of the failures. The failures are either asserts in the JIT
> 
> ` Assertion failed '!"Inconsistent profile data"' in 'Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax.Lexer:LexXmlDocCommentLeadingTrivia(byref):this' during 'Recognize Switch' (IL size 730; hash 0xe3243bf9; Tier1)`
> 
> https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-heads-main-a4b3f25a20cf46ce8d/System.Text.RegularExpressions.Tests/1/console.2ae64c75.log?skoid=8eda00af-b5ec-4be9-b69b-0919a2338892&sktid=72f988bf-86f1-41af-91ab-2d7cd011db47&skt=2024-09-24T18%3A02%3A37Z&ske=2024-09-24T19%3A02%3A37Z&sks=b&skv=2024-05-04&sv=2024-05-04&st=2024-09-24T18%3A02%3A37Z&se=2024-09-24T19%3A02%3A37Z&sr=c&sp=r&sig=e35ir71XB%2BNVh0DzOhQSt7pwGDj9BrlFxUC2QGob%2BAk%3D
> 
> Or asserts/crashes in libraries that suggest bad codegen:
> 
> https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-heads-main-5f622e8df388454594/System.Text.Json.Tests/1/console.dbf8d227.log?skoid=8eda00af-b5ec-4be9-b69b-0919a2338892&sktid=72f988bf-86f1-41af-91ab-2d7cd011db47&skt=2024-09-24T18%3A04%3A02Z&ske=2024-09-24T19%3A04%3A02Z&sks=b&skv=2024-05-04&sv=2024-05-04&st=2024-09-24T18%3A04%3A02Z&se=2024-09-24T19%3A04%3A02Z&sr=c&sp=r&sig=bL%2BAUD%2Bxfmt5N6ov%2F8w6uwftzRUy0AkdDHRY%2FSGXwbI%3D

@EgorBo, PTAL.

**Comment by jkotas** [comment]
These failures should be fixed by https://github.com/dotnet/runtime/pull/108169

**Pull request merged**